src/test/java/net/sf/jabref/model/database/BibDatabaseTest.java

```diff
@@ -317,4 +320,52 @@ public void resolveForStringsOddHashMarkAtTheEnd() {
         database.addString(string);
         assertEquals(database.resolveForStrings("AAA#AAA#AAA#"), "AAAaaaAAA#");
     }
+
+    @Test
+    public void getUsedStringsSingleStringWithString() {
+        BibEntry entry = new BibEntry(IdGenerator.next());
+        entry.setField("author", "#AAA#");
+        BibtexString string = new BibtexString(IdGenerator.next(), "AAA", "Some other #BBB#");
```

Comment:
Please move the string initalization and adding them to the db to the @setup method.